src/gc/gcinterface.d

```diff
@@ -33,6 +33,7 @@ struct Range
     void* ptop;
     TypeInfo ti; // should be tail const, but doesn't exist for references
     alias pbot this; // only consider pbot for relative ordering (opCmp)
+    bool opEquals(in Range rhs) nothrow const { return pbot == rhs.pbot; }
```

Comment:
treap uses both `==` and `<`. If the order of `alias this` and operator overload lookup for `opEquals` changes, shouldn't this also happen for `opCmp`?

Reply:
Not really, because opCmp is treated differently:

struct A
{
    int x;
    bool opCmp(ref const A q)
    {
        return this.x < q.x;                                                                                                                  
    }
}  
   
struct B
{
    A a, b;
}

struct C
{
    A a, b;
    alias a this;
}

static assert(C(A(1), A(1)) < C(A(1), A(2))); // error: need opCmp for struct C;
static assert(C(A(1), A(1)) != C(A(1), A(2))); // fails because A.opEquals is called

The dmd patch makes it so that the last assert passes (C.opEquals has a higher priority).

Reply:
>  static assert(C(A(1), A(1)) < C(A(1), A(2))); // error: need opCmp for struct C;

That just happens because your opCmp overload must not `ref` the rvalue parameter. With `    bool opCmp(A q)` I get: "Error: static assert:  C(A(1), A(1)) < C(A(1), A(2)) is false". see https://run.dlang.io/is/ZdfrfR